### PR TITLE
Add support for VEX

### PR DESCRIPTION
**Pull request by ziadhany**
image
Screenshot from 2024-04-29 20-46-55
Screenshot from 2024-05-04 23-16-35

### CODE REVIEW

**Comment by ziadhany** [comment]
@tdruez @DennisClark I'm finalizing the VCIO to CycloneDX vulnerability mapping (references, ratings, packages).  If you have any feedback, please let me know.

**Comment by DennisClark** [comment]
@ziadhany   the screen shots look nice.  Do you have an example of the VEX document that you can attach for me to review?  Thanks.

**Comment by DennisClark** [comment]
@ziadhany   Please see if you can support CycloneDX spec 1.4, 1.5, and 1.6 for the VEX.

**Comment by ziadhany** [comment]
> @ziadhany the screen shots look nice. Do you have an example of the VEX document that you can attach for me to review? Thanks.

[dejacode_nexB_product_vex(7).json](https://github.com/nexB/dejacode/files/15167725/dejacode_nexB_product_vex.7.json)

File: dejacode_toolkit/vex.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+        purl_string = affected_package.get("purl")
+
+        vul_purl = None
+        if purl_string:
+            vul_purl = PackageURL.from_string(purl_string)
+
+        versions.append(BomTargetVersionRange(version=vul_purl.version, status=status))
+
+    if versions:
+        bom_target = BomTarget(ref="urn:cdx:serialNumber/version#bom-ref", versions=versions)
```
Comment (ziadhany) [inline dejacode_toolkit/vex.py:158]: I am confused about [bom_target](https://github.com/CycloneDX/bom-examples/blob/master/VEX/vex.json#L91)
especially this part `urn:cdx:3e671687-395b-41f5-a30f-a58921a69b79/1#`
where can we get bom ref? ( vulnerablecode doesn't provide the bom ref  ) should be create a bom and pass the bom ref to vex

**Comment by DennisClark** [comment]
@ziadhany   as we agreed in our VCIO meeting, please use the Package URL (PURL) as the bom_ref.  thanks.

**Comment by tdruez** [comment]
Implemented in https://github.com/aboutcode-org/dejacode/pull/187
@ziadhany Thanks for bootstrapping this.

Closing.

**Pull request closed without merging**